### PR TITLE
[tf.learn] Remove accidentally repeated code

### LINKED ISSUE #2198: Allow manual changing a restored TensorFlowEstimator
For TensorFlowEstimator models, currently there's no way to change learning rate once the model is saved and restored.

Referencing: /tensorflow/tensorflow/contrib/learn/python/learn/estimators/base.py

### PR DESCRIPTION
**Pull request by terrytangyuan**
Tried to fix #2198 but I guess I'll fire a separate PR when I get a chance

### CODE REVIEW

**Comment by petewarden** [comment]
Test this please, Jenkins.

**Comment by Kaizenathon** [comment]
omg im so noob

File: tensorflow/contrib/learn/python/learn/estimators/base.py [hunk 1]
```diff
@@ -919,17 +919,19 @@ def _restore(self, path):
 
   # pylint: disable=unused-argument
   @classmethod
-  def restore(cls, path, config=None):
+  def restore(cls, path, config=None, new_params=None):
```
Comment (vrv) [inline tensorflow/contrib/learn/python/learn/estimators/base.py]: you should probably add some tests

Comment (terrytangyuan) [inline tensorflow/contrib/learn/python/learn/estimators/base.py]: Sure, done.

Comment (terrytangyuan) [inline tensorflow/contrib/learn/python/learn/estimators/base.py]: @vrv

**Comment by terrytangyuan** [comment]
@petewarden @vrv Could any of you run the Jenkins again? I just found a huge portion of repeated code - maybe from conflicts not being resolved correctly during internal branch merge. Also cc @ilblackdragon

File: tensorflow/contrib/learn/python/learn/estimators/base.py [hunk 2]
```diff
@@ -365,429 +365,6 @@ def get_tensor(self, name):
     """
     return self._graph.get_tensor_by_name(name)
 
-    def __init__(self, model_fn, n_classes, batch_size=32,
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/estimators/base.py]: where did all this go?

File: tensorflow/contrib/learn/python/learn/tests/test_saver.py [hunk 3]
```diff
@@ -83,6 +83,20 @@ def testNoCheckpoints(self):
...
+    classifier = learn.TensorFlowDNNClassifier(hidden_units=[10, 20, 10],
+                                               n_classes=3,
+                                               learning_rate=0.1)
+    classifier.fit(iris.data, iris.target)
+    classifier.save(path)
+    self.assertEqual(classifier.learning_rate, 0.1)
+    new_classifier = learn.TensorFlowEstimator.restore(path,
+      new_params={'learning_rate': 0.2})
+    self.assertEqual(type(new_classifier), type(classifier))
+    self.assertEqual(new_classifier.learning_rate, 0.2)
```
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/tests/test_saver.py]: this not going to work the way you expect. for example learning rate is now captured in the variable - so resetting it on the class - won't help at this point to get a different learning rate.

We are moving an Estimator class that uses graph_actions, which re-creates graph each time `fit`/`predict` is called. That way one will be able to override parameters.

File: tensorflow/contrib/learn/python/learn/estimators/base.py (hunk 2 above)
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/estimators/base.py]: oh, ic - this is repeated code. how do test pass with this?...

Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/estimators/base.py]: ic, it's repeated chunk. how did even tests pass?...

File: tensorflow/contrib/learn/python/learn/tests/test_saver.py (hunk 3 above)
Comment (ilblackdragon) [inline tensorflow/contrib/learn/python/learn/tests/test_saver.py]: that said - there is a lot of other arguments changing will lead to problems, for example if one changes hidden_units or n_classes - it will lead to a mismatch with checkpoints and etc.

I would instead recommend creating new Estimator with new params and then just restoring model variables from existing checkpoints. see `learn/utils/checkpoints` library for an `init_from_checkpoints` function that can help with loading. We can add a nice interface for it on Estimator (not sure how it should look yet though), but already one can use it to restore parameters from another trained estimator.

**Comment by terrytangyuan** [comment]
@ilblackdragon Good to know. I'll take a look at graph_actions later when I get a chance. I've reverted the changes. Let's just keep this PR to be removing those repeated lines (tests passed since the repeated code are inside a function block). Could you take a look and merge this first or people will have some issues?

**Comment by ilblackdragon** [comment]
@tensorflow-jenkins test this please.
